server/src/main/java/org/elasticsearch/http/HttpTracer.java

```diff
@@ -57,12 +57,13 @@ HttpTracer maybeLogRequest(RestRequest restRequest, @Nullable Exception e) {
         if (logger.isTraceEnabled() && TransportService.shouldTraceAction(restRequest.uri(), tracerLogInclude, tracerLogExclude)) {
             logger.trace(
                 () -> format(
-                    "[%s][%s][%s][%s] received request from [%s]",
+                    "[%s][%s][%s][%s] received request from [%s]%s",
                     restRequest.getRequestId(),
                     restRequest.header(Task.X_OPAQUE_ID_HTTP_HEADER),
                     restRequest.method(),
                     restRequest.uri(),
-                    restRequest.getHttpChannel()
+                    restRequest.getHttpChannel(),
+                    Task.extractTraceId(restRequest.header(Task.TRACE_PARENT_HTTP_HEADER)).map(t -> " trace.id: " + t).orElse("")
```

Comment:
Can we add a test ensuring the new code produces the expected result? I know logger is hard to test, so perhaps we can extract the code into a method? I also recall we had a way to ensure logs had a certain message in some tests, but I can't seem to find it now.

Reply:
Unfortunately that's in a different package, and `HttpTracer` is package private. I'll see what I can do

Reply:
I've added a separate test